### ISSUE TITLE
What happens if you run the installation of the miner again

### ISSUE THREAD

**Issue by RHGF44** [UnamSanctam/SilentCryptoMiner#1626]
Hello.
What happens if you run the installation of the miner again? Will the old one be deleted and the new one launched?
Will the accumulated hashes of the old one be deleted?

**Comment by UnamSanctam** [comment]
It will install the new file over the old file, but since the miner is already running it will only replace the file and then close (it doesn't need to inject the miner since it's already running). Not fully certain what you mean by accumulated hashes, do you mean the found shares submitted to the pool? Then no the pool handles that, if you just mean the total hashes checked in the miner then that doesn't matter since the total hashes on the miner side is purely statistical and nothing else. Since finding a share is based on probability then the total hashes does not matter (you have the same chance of finding a share after 10 hashes or 10 000 000 hashes).

**Comment by RHGF44** [comment]
That is, when you restart, the work of the miner does not stop?

**Comment by RHGF44** [comment]
When restarting the computer, the mining process is suspended, and when you start the re-installation, it is suspended?

**Comment by UnamSanctam** [comment]
No, if you re-install it then it only does the file replacement and nothing more, since the miner is already running.

**Comment by RHGF44** [comment]
Thank you

**Comment by RHGF44** [comment]
And if you reinstall the miner with a different configuration, what will happen? When will the configuration change?

**Comment by UnamSanctam** [comment]
If you install a completely new miner then both will be installed, if you have the same "Startup" configuration as the old miner then the new miner will replace the old miners file, though the old miner itself will still be running until the next restart. That's because the miner is hidden, so the new miner cannot (technically) find the old miner.

**Comment by RHGF44** [comment]
By configuration, I mean only the name of the worker and the address of the pool. If I start a new miner with these changes, will it run in parallel until the computer is restarted?

**Comment by UnamSanctam** [comment]
Do you mean creating a new miner with a different worker name and address or by changing it through a "Remote Configuration" or web panel? If it's a new miner then it will run in parallel since every build can only be found by itself (no matter the configuration). If the configuration is changed using a "Remote Configuration" or web panel then it will just change the current miner of course.

**Comment by RHGF44** [comment]
With the configuration removed, it is clear that it will not replace the miner. I've been talking about starting the installer with a new configuration from the beginning.

**Comment by RHGF44** [comment]
If the same miner installer is run again, what will happen? (how it will affect the work of the miner launched for the first time)

**Comment by UnamSanctam** [comment]
If you run the same installer then it will only replace the file and then close since that installers miner is already running.

**Comment by RHGF44** [comment]
What file will it replace?

**Comment by RHGF44** [comment]
> Yes if you restart the miner or the computer (with "Startup" enabled) then the miner will continue mining as before.

1. What does this mean? (with "Startup" enabled)
2. What setting in the miner sets this up?

**Comment by UnamSanctam** [comment]
"Startup" is the setting for enabling installation.
You enable it inside the "Startup" tab.

**Comment by RHGF44** [comment]
Can I add or change the address of the control panel via remote configuration? (in json)

**Comment by UnamSanctam** [comment]
Yes you can using the `"api-endpoint": "API Endpoint URL"` option.
